blocks/bricks/sequence_generators.py

```diff
@@ -714,8 +714,9 @@ def apply_delegate(self):
         # I can write self.apply because it can be overriden.
         # Thus I have to hack.
         # TODO: nice interface for this trick.
-        AttentionTransition.do_apply.__get__(self, None)
-        return AttentionTransition.do_apply
+        # Bart: This doesn't work anymore, and probably shouldn't; without a
+        # brick attached, it's impossible to call do_apply(brick)...
+        return self.do_apply
```

Comment:
You're absolutely right. It was in the version of my code that I didn't go with in the end that this didn't work anymore. I reintroduced it in #127, and also made the interface work as you suggested.